File: data/recomposed_projects/gpt-4o-2024-11-20/body/0.0/commons-fileupload/src/main/org/apache/commons/fileupload/FileUploadException.py
from __future__ import annotations
import re
from io import StringIO
import io
import typing
from typing import *


class FileUploadException(Exception):

    __cause: BaseException = None

    __serialVersionUID: int = 8881893724388807504

    def getCause(self) -> BaseException:
        return self.__cause

    def printStackTrace1(
        self, writer: typing.Union[io.TextIOWrapper, io.StringIO]
    ) -> None:
        super().printStack(writer)
        if self.__cause is not None:
            writer.write("Caused by:\n")
            self.__cause.printStack(writer)

    def printStackTrace0(self, stream: typing.IO) -> None:
        super().printStack(stream)
        if self.__cause is not None:
            stream.write("Caused by:\n")
            self.__cause.printStack(stream)

    def __init__(self, msg: str, cause: BaseException) -> None:
        super().__init__(msg)
        self.__cause = cause

    @staticmethod
    def FileUploadException1(msg: str) -> FileUploadException:
        return FileUploadException(msg, None)

    @staticmethod
    def FileUploadException0() -> FileUploadException:
        return FileUploadException(None, None)

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/FileUploadException.java
package org.apache.commons.fileupload;

import java.io.PrintStream;
import java.io.PrintWriter;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class FileUploadException extends Exception {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/FileUploadException.py", "FileUploadException"));
  public static final long serialVersionUID = 8881893724388807504L;
  public Throwable cause;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  @Override
  public Throwable getCause() {

    return cause;
  }

  public void printStackTrace1(PrintWriter writer) {

    super.printStackTrace(writer);
    if (cause != null) {
      writer.println("Caused by:");
      cause.printStackTrace(writer);
    }
  }

  public void printStackTrace0(PrintStream stream) {

    super.printStackTrace(stream);
    if (cause != null) {
      stream.println("Caused by:");
      cause.printStackTrace(stream);
    }
  }

  public FileUploadException(String msg, Throwable cause) {

    super(msg);
    this.cause = cause;
  }

  public static FileUploadException FileUploadException1(final String msg) {

    return new FileUploadException(msg, null);
  }

  public static FileUploadException FileUploadException0() {

    return new FileUploadException(null, null);
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {

    this.cause =
        (Throwable)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_FileUploadException__cause"), "Throwable", idMap, cause);
    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {

    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_FileUploadException__cause",
        IntegrationUtils.mapToPython(
            cause, idMap, this.obj.getMember("_FileUploadException__cause")));
    return idMap;
  }
}
